tests/util/addClass.test.js

```diff
@@ -1,28 +1,54 @@
 import addClass from "../../src/util/addClass";
 
-describe('addClass', () => {
-    test('should append when className is empty', () => {
-      const el = document.createElement('div');
-      addClass(el, 'myClass');
-      expect(el.className).toBe('myClass');
-    });
+describe("addClass", () => {
+  test("should append when className is empty", () => {
+    const el = document.createElement("div");
+    addClass(el, "myClass");
+    expect(el.className).toBe("myClass");
+  });
 
-    test('should append when className is NOT empty', () => {
-      const el = document.createElement('div');
-      el.className = 'firstClass';
+  test("should append when className is NOT empty", () => {
+    const el = document.createElement("div");
+    el.className = "firstClass";
 
-      addClass(el, 'secondClass');
+    addClass(el, "secondClass");
 
-      expect(el.className).toBe('firstClass secondClass');
-    });
+    expect(el.className).toBe("firstClass secondClass");
+  });
 
-    test('should append when element is SVG', () => {
-      const el = document.createElement('svg');
-      el.className = 'firstClass';
+  test("should append when element is SVG", () => {
+    const el = document.createElement("svg");
+    el.className = "firstClass";
 
-      addClass(el, 'secondClass');
+    addClass(el, "secondClass");
 
-      expect(el.className).toBe('firstClass secondClass');
-    });
-  }
-);
+    expect(el.className).toBe("firstClass secondClass");
+  });
+
+  test("should not append duplicate classNames to svg elements", () => {
+    const el = document.createElement("svg");
+    el.className = "firstClass";
+
+    addClass(el, "firstClass");
+
+    expect(el.className).toBe("firstClass");
+  });
+
+  test("should not append duplicate classNames to elements", () => {
+    const el = document.createElement("div");
+    el.className = "firstClass";
+
+    addClass(el, "firstClass");
+
+    expect(el.className).toBe("firstClass");
+  });
+
+  test("should not append duplicate list of classNames to elements", () => {
+    const el = document.createElement("div");
+    el.className = "firstClass firstClass";
+
+    addClass(el, "firstClass firstClass firstClass");
+
+    expect(el.className).toBe("firstClass");
```

Comment:
a very quirky test; I expect this isn't always true.  Only true when `elem.classList !== undefined`; otherwise:

```js
expect(el.className).toBe('firstClass firstClass firstClass firstClass firstClass');
```

but, probably not important. 😆

Reply:
just added a new test for empty classname

Reply:
Probably not important; I was just saying that for (I guess) IE9 the quirky test case would be:

```javascript
test("does append duplicate list of classNames if element does not support classList", () => {
    const el = { className: '' }; // fake pseudo-element
    const classes = "firstClass firstClass firstClass";

    addClass(el, classes);

    expect(el.className).toBe(" " + classes);
  });
```

Again, doesn't really matter.  Might be worth removing that portion of addClass, since I'm not aware of anyone using IE9 anymore.